### PR TITLE
Add db_index option to Redis cache configuration

### PR DESCRIPTION
**Pull request by DeanNeaht**
## Description

This PR introduces the db_index option in the Redis cache configuration for chproxy. The feature allows specifying a particular Redis database index for caching

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Checklist

- [X] Linter passes correctly
- [X] Add tests which fail without the change (if possible)
- [X] All tests passing
- [X] Extended the README / documentation, if necessary

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by mga-chka** [comment]
FYI all the maintainers are off, we'll review the PR in January

File: config/README.md
Comment (mga-chka) [inline config/README.md]: Hi, can you add a comment that it only works for non clustered redis? Even its "obivous" for a redis poweruser, it might prevent some false issues on the chproxy repo.

**Review by mga-chka — APPROVED** [review]
Please check my comment.
Once its done I'll merge you PR and it will be part of the next release (that should happen in 1-2 weeks, the time to have all the current PR approved & merged)

File: config/README.md
Comment (DeanNeaht) [reply config/README.md]: Fixed

**Pull request merged**
